server/database/models/pub.js

```diff
@@ -4,10 +4,9 @@ mongoose.promise = Promise;
 
 // define pubSchema
 const pubSchema = new Schema({
-    name: { type: String, unique: false, required: false },
-    twitterHandle: { type: String, unique: true, required: false }
-})
-
+  name: { type: String, unique: false, required: false },
+  twitterHandle: { type: String, unique: true, sparse: true, required: false },
```

Comment:
@LGG233 I ran into an issue creating multiple new users. When a new user is created, they have zero publications. Mongo barfs some "duplicate key" error on `twitterHandle` since it is `unique: true`. This is because Mongo actually sets `twitterHandle = null` behind the scenes. Two twitter handles = null would not be unique! The workaround here is to set the `sparse: true` flag to essentially ignore `null/undefined` values when checking for uniqueness.

Reply:
Nice fix!